redis/src/main/scala/zio/redis/ResultBuilder.scala

```diff
@@ -1,19 +1,44 @@
 package zio.redis
 
 import zio.ZIO
+import zio.redis.ResultBuilder.NeedsReturnType
 import zio.schema.Schema
 
-trait ResultBuilder[+F[+_]] {
-  import ResultBuilder.NeedsReturnType
-
-  def returning[R: Schema]: ZIO[RedisExecutor, RedisError, F[R]]
+sealed trait ResultBuilder {
 
+  /**
+   * The purpose of method to give a hint what is the error if you don't call `returning` method
+   */
   final def map(f: Nothing => Any)(implicit nrt: NeedsReturnType): ZIO[Any, Nothing, Nothing] = ???
 
+  /**
+   * The purpose of method to give a hint what is the error if you don't call `returning` method
+   */
   final def flatMap(f: Nothing => Any)(implicit nrt: NeedsReturnType): ZIO[Any, Nothing, Nothing] = ???
 }
 
 object ResultBuilder {
+
   @annotation.implicitNotFound("Use `returning[A]` to specify method's return type")
   final abstract class NeedsReturnType
+
+  trait ResultSchemaBuilder extends ResultBuilder {
+    def returning[R: Schema]: ZIO[RedisExecutor, RedisError, R]
+  }
+
+  trait ResultSchemaBuilder1[F[_]] extends ResultBuilder {
+    def returning[R: Schema]: ZIO[RedisExecutor, RedisError, F[R]]
+  }
+
+  trait ResultSchemaBuilder2[F[_, _]] extends ResultBuilder {
+    def returning[R1: Schema, R2: Schema]: ZIO[RedisExecutor, RedisError, F[R1, R2]]
+  }
+
+  trait ResultSchemaBuilder3[F[_, _, _]] extends ResultBuilder {
+    def returning[R1: Schema, R2: Schema, R3: Schema]: ZIO[RedisExecutor, RedisError, F[R1, R2, R3]]
+  }
```

Comment:
There should be some room for variance here and there (e.g. what you named `ResultSchemaBuidler1` was covariant), and I'd definitely remove `Schema` from their names.

Reply:
Add covariance to F, should I return covariance to underscores?
I don't understand how that suppose to be useful in our case. I prefer not to do what i don't understand how is going to be used and to do as easy as possible)) that's why i deleted it. 
If you can explain me that i appreciate it.